src/npctalk_funcs.cpp

```diff
@@ -499,6 +500,50 @@ void talk_function::give_all_aid( npc &p )
     }
 }
 
+static void generic_barber( std::string mut_type )
+{
+    uilist hair_menu;
+    std::string menu_text = "";
+    if( mut_type == "hair_style" ) {
+        menu_text = _( "Choose a new hairstyle" );
+    } else if( mut_type == "facial_hair" ) {
+        menu_text = _( "Choose a new facial hair style" );
+    }
+    hair_menu.text = menu_text;
+    int index = 0;
+    hair_menu.addentry( index, true, 'q', _( "Actually... I've changed my mind." ) );
+    std::vector<trait_id> hair_muts = get_mutations_in_type( mut_type );
+    trait_id cur_hair;
+    for( auto elem : hair_muts ) {
+        if( g->u.has_trait( elem ) ) {
+            cur_hair = elem;
+        }
+        index += 1;
+        hair_menu.addentry( index, true, MENU_AUTOASSIGN, elem.obj().name() );
+    }
+    hair_menu.query();
+    int choice = hair_menu.ret;
+    if( choice != 0 ) {
+        if( g->u.has_trait( cur_hair ) ) {
+            g->u.remove_mutation( cur_hair, true );
+            g->u.set_mutation( hair_muts[ choice - 1 ] );
+            add_msg( m_info, _( "You get a trendy new cut!" ) );
+        }
```

Comment:
Even if player chooses something in the menu, it will do nothing in case he doesn't have any other cut?

Reply:
Yeah changed that to always do the cut, but remove cur_hair only if it was set.